storage/intent_resolver.go

```diff
@@ -417,11 +423,16 @@ func (ir *intentResolver) resolveIntents(ctx context.Context, r *Replica,
 			return pErr.GoError()
 		}
 		wg.Add(1)
-		if wait || !r.store.Stopper().RunLimitedAsyncTask(ir.sem, func() {
-			if err := action(); err != nil {
-				log.Warningf("unable to resolve local intents; %s", err)
-			}
-		}) {
+		var runErr error
+		if !wait {
+			runErr = r.store.Stopper().RunLimitedAsyncTask(ir.sem, func() {
+				if err := action(); err != nil {
+					log.Warningf("unable to resolve local intents; %s", err)
+				}
+			})
+		}
+
+		if wait || runErr != nil {
```

Comment:
While you're here, add a comment to the block below that it's probably ripe for removal (including running the task).